modules/saveload/saveload_api.h

```diff
@@ -0,0 +1,86 @@
+/**************************************************************************/
+/*  saveload_api.h                                                        */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef SAVELOAD_API_H
+#define SAVELOAD_API_H
+
+#include "core/object/ref_counted.h"
+
+class SaveloadAPI : public RefCounted {
+	GDCLASS(SaveloadAPI, RefCounted);
+
+	static SaveloadAPI *singleton;
```

Comment:
It being `RefCounted` and having a singleton are mutually exclusive, see `MultiplayerAPI`

Reply:
Thanks for the tip! I figured as much from reading through the error logs, but it's nice to have confirmation that my suspicion was right :)

Reply:
@AThousandShips In changing from RefCounted to Object, will I have to add some logic to free `SaveloadAPI` to avoid a memory leak?

All I have for the destructor right now is

```cpp
virtual ~SaveloadAPI() {}
```

Reply:
Look at how it's done in other modules on register/unregister, like navigation with the server there